docker/generate_makefile.py

```diff
@@ -4,200 +4,140 @@
-
-.{fuzzer}-{benchmark}-builder: .{fuzzer}-{benchmark}-builder-intermediate
-	docker build \\
-    --tag {base_tag}/builders/{fuzzer}/{benchmark} \\
-    --file=docker/benchmark-builder/Dockerfile \\
-    --build-arg parent_image={base_tag}/builders/{fuzzer}/{benchmark}-intermediate \\
-    --build-arg fuzzer={fuzzer} \\
-    --build-arg benchmark={benchmark} \\
-    $(call cache_from,{base_tag}/builders/{fuzzer}/{benchmark}) \\
-    .
-
-.pull-{fuzzer}-{benchmark}-builder: .pull-{fuzzer}-{benchmark}-builder-intermediate
-	docker pull {base_tag}/builders/{fuzzer}/{benchmark}
-
-ifneq ({fuzzer}, coverage)
-
-.{fuzzer}-{benchmark}-intermediate-runner: base-runner
-	docker build \\
-    --tag {base_tag}/runners/{fuzzer}/{benchmark}-intermediate \\
-    --file fuzzers/{fuzzer}/runner.Dockerfile \\
-    $(call cache_from,{base_tag}/runners/{fuzzer}/{benchmark}-intermediate) \\
-    fuzzers/{fuzzer}
-
-.pull-{fuzzer}-{benchmark}-intermediate-runner: pull-base-runner
-	docker pull {base_tag}/runners/{fuzzer}/{benchmark}-intermediate
-
-.{fuzzer}-{benchmark}-runner: .{fuzzer}-{benchmark}-builder .{fuzzer}-{benchmark}-intermediate-runner
-	docker build \\
-    --tag {base_tag}/runners/{fuzzer}/{benchmark} \\
-    --build-arg fuzzer={fuzzer} \\
-    --build-arg benchmark={benchmark} \\
-    $(call cache_from,{base_tag}/runners/{fuzzer}/{benchmark}) \\
-    --file docker/benchmark-runner/Dockerfile \\
-    .
-
-.pull-{fuzzer}-{benchmark}-runner: .pull-{fuzzer}-{benchmark}-builder .pull-{fuzzer}-{benchmark}-intermediate-runner
-	docker pull {base_tag}/runners/{fuzzer}/{benchmark}
-
-""" + FUZZER_BENCHMARK_RUN_TARGETS_TEMPLATE + """
-
-else
-
-build-{fuzzer}-{benchmark}: .{fuzzer}-{benchmark}-builder
-pull-{fuzzer}-{benchmark}: .pull-{fuzzer}-{benchmark}-builder
-
-endif
-"""
-
-
-def generate_fuzzer(fuzzer, benchmarks):
-    """Output make rules for a single fuzzer."""
-    # Generate build rules for the fuzzer itself.
-    print(FUZZER_TEMPLATE.format(fuzzer=fuzzer, base_tag=BASE_TAG))
-
-    # Generate rules for fuzzer-benchmark pairs.
+from common import yaml_utils
+from common import benchmark_utils
+from common import fuzzer_utils
+from experiment.build import docker_images
+
+BASE_TAG = "gcr.io/fuzzbench"
+BENCHMARK_DIR = benchmark_utils.BENCHMARKS_DIR
+
+RUN_TEMPLATE = """
+{run_type}-{fuzzer}-{benchmark}: .{fuzzer}-{benchmark}-runner
+\tdocker run \\
+\t--cpus=1 \\
+\t--cap-add SYS_NICE \\
+\t--cap-add SYS_PTRACE \\
+\t-e FUZZ_OUTSIDE_EXPERIMENT=1 \\
+\t-e FORCE_LOCAL=1 \\
+\t-e TRIAL_ID=1 \\
+\t-e FUZZER={fuzzer} \\
+\t-e BENCHMARK={benchmark} \\"""
+
+
+def print_benchmark_definition(benchmarks):
+    """Prints benchmark variables from benchmark.yaml files."""
     for benchmark in benchmarks:
+        benchmark_vars = yaml_utils.read(
+            os.path.join(BENCHMARK_DIR, benchmark, 'benchmark.yaml'))
+        print(benchmark + '-fuzz-target=' + benchmark_vars['fuzz_target'])
+        print()
```

Comment:
i don't get why we need to provide the FUZZ_TARGET=$({benchmark}-fuzz-target) to run a benchmark. shouldn't the `fuzzer.py` know how to run a benchmark? @jonathanmetzman

Reply:
It needs this because it doesn't use the benchmark.yaml (which specifies the name of the target). Some project builds will have multiple targets so we need to know which to use.